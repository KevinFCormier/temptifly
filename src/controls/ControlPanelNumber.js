'use strict'

import React from 'react'
import PropTypes from 'prop-types'
import { FormGroup, Popover } from '@patternfly/react-core'
import HelpIcon from '@patternfly/react-icons/dist/js/icons/help-icon'

class ControlPanelNumber extends React.Component {
  static propTypes = {
    control: PropTypes.object,
    controlId: PropTypes.string,
    handleChange: PropTypes.func,
    i18n: PropTypes.func
  };

  constructor(props) {
    super(props)
    this.state = {}
  }

  setControlRef = (control, ref) => {
    control.ref = ref
  };

  render() {
    const { controlId, i18n, control, handleChange } = this.props
    const { name, initial, exception, validation={}, tooltip } = control
    let { active } = control
    active = active || initial
    const nActive = parseInt(active, 10)

    const onChange = (value) => {
      control.active = value.toString()
      handleChange()
    }

    const validated = exception ? 'error' : undefined
    return (
      <React.Fragment>
        <div
          className="creation-view-controls-number"
          ref={this.setControlRef.bind(this, control)}
        >
          <FormGroup
            id={`${controlId}-label`}
            label={name}
            isRequired={validation.required}
            fieldId={controlId}
            helperTextInvalid={exception}
            validated={validated}
            labelIcon={
              /* istanbul ignore next */
              tooltip ? (
                <Popover
                  id={`${controlId}-label-help-popover`}
                  bodyContent={tooltip}
                >
                  <button
                    id={`${controlId}-label-help-button`}
                    aria-label="More info"
                    onClick={(e) => e.preventDefault()}
                    className="pf-c-form__group-label-help"
                  >
                    <HelpIcon noVerticalAlign />
                  </button>
                </Popover>
              ) : (
                <React.Fragment />
              )
            }
          >
            <div className="pf-c-number-input">
              <div className="pf-c-input-group">
                <button className="pf-c-button pf-m-control" style={{lineHeight: '16px'}} type="button" aria-label="Minus"
                  onClick={()=>{
                    onChange(nActive-1)
                  }}>
                  <span className="pf-c-number-input__icon">
                    <svg height="16" width="16" role="img" viewBox="0 0 24 24">
                      <path d="M0 10h24v4h-24z" />
                    </svg>
                  </span>
                </button>
                <input
                  className="pf-c-form-control"
                  type="number"
                  value={active}
                  name="number-input-default-name"
                  onFocus={e => {
                    e.target.select()
                  }}
                  onChange={e => {
                    onChange(e.target.value)
                  }}
                  aria-label="Number input"
                />
                <button className="pf-c-button pf-m-control" style={{lineHeight: '16px'}} type="button" aria-label="Plus"
                  onClick={()=>{
                    onChange(nActive+1)
                  }}>
                  <svg height="16" width="16" role="img" viewBox="0 0 24 24">
                    <path d="M24 10h-10v-10h-4v10h-10v4h10v10h4v-10h10z" />
                  </svg>
                </button>
              </div>
            </div>
            {validated === 'error' ? (
              <div style={{ borderTop: '1.75px solid red', paddingBottom: '6px', maxWidth: '400px' }}></div>
            ) : (
              <React.Fragment />
            )}
          </FormGroup>
        </div>
      </React.Fragment>
    )
  }

}

export default ControlPanelNumber
