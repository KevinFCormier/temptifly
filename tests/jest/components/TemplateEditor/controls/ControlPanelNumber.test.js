'use strict'

import React from 'react'
import ControlPanelNumber from '../../../../../src/controls/ControlPanelNumber'
import renderer from 'react-test-renderer'

import { control } from '../../TestingData'

describe('ControlPanelNumber component', () => {
  it('renders as expected', () => {
    const fn = jest.fn()
    const component = renderer.create(
      <ControlPanelNumber
        key={'key'}
        control={control}
        controlId={'controlId'}
        handleChange={fn}
        i18n={fn}
      />
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
})
