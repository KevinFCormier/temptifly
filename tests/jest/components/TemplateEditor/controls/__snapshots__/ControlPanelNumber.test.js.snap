// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ControlPanelNumber component renders as expected 1`] = `
<div
  className="creation-view-controls-number"
>
  <label
    className="creation-view-controls-multiselect-title"
    htmlFor="controlId"
  >
    creation.app.name
    <div
      className="bx--tooltip__label"
    >
      
      <div
        aria-expanded={false}
        aria-haspopup="true"
        aria-label="tooltip"
        className="bx--tooltip__trigger"
        id=""
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        role="button"
        tabIndex={0}
        title=""
      >
        <svg
          alt="tooltip"
          aria-label="tooltip"
          className={undefined}
          fill={undefined}
          fillRule="evenodd"
          height="16"
          name={undefined}
          role="img"
          style={undefined}
          viewBox="0 0 16 16"
          width="16"
        >
          <title>
            
          </title>
          <path
            d="M8,14.5 C11.5898509,14.5 14.5,11.5898509 14.5,8 C14.5,4.41014913 11.5898509,1.5 8,1.5 C4.41014913,1.5 1.5,4.41014913 1.5,8 C1.5,11.5898509 4.41014913,14.5 8,14.5 Z M8,16 C3.581722,16 0,12.418278 0,8 C0,3.581722 3.581722,0 8,0 C12.418278,0 16,3.581722 16,8 C16,12.418278 12.418278,16 8,16 Z"
          />
          <polygon
            points="9 13 7 13 7 7 9 7"
          />
          <circle
            cx="8"
            cy="4"
            r="1"
          />
        </svg>
      </div>
    </div>
  </label>
  <div
    className="bx--form-item"
  >
    <div
      className="bx--number bx--number--helpertext"
    >
      <div
        className="bx--number__controls"
      >
        <button
          aria-atomic="true"
          aria-label="Increment number"
          aria-live="polite"
          className="bx--number__control-btn up-icon"
          disabled={false}
          onClick={[Function]}
          title="Increment number"
          type="button"
        >
          <svg
            alt="choose a number"
            aria-label="choose a number"
            className="up-icon"
            fill={undefined}
            fillRule="evenodd"
            height="5"
            name={undefined}
            role="img"
            style={undefined}
            viewBox="0 0 10 5"
            width="10"
          >
            <title>
              choose a number
            </title>
            <path
              d="M0 5L5 .002 10 5z"
            />
          </svg>
        </button>
        <button
          aria-atomic="true"
          aria-label="Decrement number"
          aria-live="polite"
          className="bx--number__control-btn down-icon"
          disabled={false}
          onClick={[Function]}
          title="Decrement number"
          type="button"
        >
          <svg
            alt="choose a number"
            aria-label="choose a number"
            className="down-icon"
            fill={undefined}
            fillRule="evenodd"
            height="5"
            name={undefined}
            role="img"
            style={undefined}
            viewBox="0 0 10 5"
            width="10"
          >
            <title>
              choose a number
            </title>
            <path
              d="M0 0l5 4.998L10 0z"
            />
          </svg>
        </button>
      </div>
      <label
        className="bx--label"
        htmlFor="controlId"
      >
         
      </label>
      <input
        aria-describedby="controlId-error-id"
        aria-invalid={null}
        data-invalid={null}
        disabled={false}
        id="controlId"
        max={100}
        min={1}
        onChange={[Function]}
        onClick={[Function]}
        pattern="[0-9]*"
        step={1}
        type="number"
        value={1}
      />
    </div>
  </div>
</div>
`;
